mods/persistence/modules/world_save/serializers/json_serializer.dm

```diff
@@ -31,8 +31,10 @@
 				results[V] = "FLAT_OBJ#[SerializeDatum(VV)]"
 			else
 				results[V] = "OBJ#[sql.SerializeDatum(VV)]"
+	// If the object is a wrapper, keep track of the reference to the wrapped object rather than the wrapper itself.
+	var/thing_id = object_parent ? "\ref[object_parent]" : "\ref[object]"
```

Comment:
What?

This seems anomalous, why is this happening here and what is it supposed to do? Later thing_id is used for the reverse mapper, but \ref[] returns a memory address, and all entries in reverse_map are integer-strings. This breaks convention.

Reply:
The use of \ref[] here is arbitrary; it's just to get a unique identifier for the datum to ensure that only one wrapper per object is created on de-serialization (which itself has issues, but in general this approach was a bit makeshift). The issue that arises is that because flattened objects are not otherwise saved, there isn't any index to use for the key.

Needless to say though, this is pretty lackluster even if it 'works'. A possible alternative would be to add another table which contains the wrappers, and change the JSON serializer to use thing_map to make sure that each reference to an object only points to the single wrapper. I have next to 0 experience with SQL or databases in general however, and I'm not sure of the potential performance impact this would have as opposed to other solutions.